Reader tag pages: update column styles

creativecoder opened the pull request:
Related to https://github.com/Automattic/wp-calypso/pull/75730

## Proposed Changes

Updates the column styles on the newly designed reader tag pages to better match the design.

Additional markup changes to support the design include:

- Adding a `streamHeader` prop to `ReaderStream`, so a custom header can be added at the top of the main column in the two-column view, rather than above both columns.
- Modifies `ReaderStream` so the `.stream__two-column` class is only present when two columns are displayed side-by-side, for styles that only apply to the two column layout.
- Adds the `.tag-stream__main` class to the `main` element on the tag page to more easily customize the tag page (it appears this was present before, but removed at some point).

<img width="1353" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1699996/13b6ca46-3ca7-4958-a22d-7a0e3ab00c97">

## Testing Instructions

* Visit `/tag/{tag-slug}` and compare with designs (pe7F0s-IS-p2)
* Check both large (two column) and small screens (tabbed layout)
* Spot check other Reader pages: Following, Search, etc to check for regressions

Commit:
Moves tag stream header to fix alignment issue

Commit:
Updates tag page column styles from design

Commit:
Set max width for search input in sidebar

Commit:
Adjust sidebar border

Commit:
Account for prompt tag pages with descriptions

creativecoder commented on client/reader/stream/style.scss:
```diff
@@ -655,98 +652,93 @@
 		line-height: 18px;
 	}
 
-	.stream__right-column {
```
At this point and below for this file, there are no css property changes.

Several blocks are being moved from `.stream__two-column .stream__right-column` to `.stream__right-column` so the styles apply to the right column on both two column and tabbed views.

Addison-Stavlo reviewed (commented):
This is looking pretty good! 

In testing the first thing I noticed was the position of the 'back' button in mobile view gets uncomfortably close to the title. However, this seems to be pre-existing and can probably be addressed separately: 

<img width="187" alt="Screenshot 2023-05-15 at 8 45 44 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/6fc68144-9395-414b-a5f3-4e69f8b2e19a">

<hr/> 

> Spot check other Reader pages: Following, Search, etc to check for regressions 


WRT this, i found a few differences between running this branch and production. Most seem pretty minor, but noting them blow:

1. The images seem to extend outside the column slightly on this branch, both for dual-column and tab views of the search page. 

![reader-search-columns](https://github.com/Automattic/wp-calypso/assets/28742426/533279db-0bfa-47e4-bde5-1eae9d4c3d79)

2. Items in the right column are not concatenated as early / at as short of a length (this seems like an improvement!). Its especially noticeable on my long titled test sites:

![less-concat](https://github.com/Automattic/wp-calypso/assets/28742426/6f654b9b-fcff-40b0-917d-75bc52c4c18a)

3. There seems to be a different font size/colors for subheading links on the likes (and conversations) pages:


![fonts-diff](https://github.com/Automattic/wp-calypso/assets/28742426/b8c1070d-7b21-443f-b142-6364212e0bfa)

Commit:
Address style issues

- Restore `reader__content` class on single column reader pages
- Remove margin on search pages
- Use default position for Back link on tag pages

creativecoder commented:
@Addison-Stavlo Thanks for the detailed testing/review!

> the 'back' button in mobile view gets uncomfortably close to the title.

I went ahead and removed the repositioning styles for various breakpoints--it seems to look fine with the default positioning.

> The images seem to extend outside the column slightly on this branch, both for dual-column and tab views of the search page.

You're right--there was padding and margin inside the container that doesn't seem necessary; the design looks to be flush with the column. So I've removed it.

> Items in the right column are not concatenated as early / at as short of a length

Yes, that was intentional!

> There seems to be a different font size/colors for subheading links on the likes (and conversations) pages

Thanks for catching that! I missed one place that needed the `reader__content` class now that I moved it one level above the `InfiniteList` component. Should now be resolved.

---

I'd appreciate another look when you have a moment.
